doc/sphinx/notebooks/advanced/varying_noise.pct.py

```diff
@@ -24,284 +24,314 @@
+# ### Generate data
 # We create a utility function to generate synthetic data, including noise that varies amongst the data:
 
 # %%
-np.random.seed(1)  # for reproducibility
+def generate_data():
+    rng = np.random.default_rng(42)  # for reproducibility
+    X = rng.uniform(0.0, 1.0, (n_data, 1))
+    signal = (X - 0.5) ** 2 + 0.05
+    noise_scale = 0.5 * signal
+    noise = noise_scale * rng.standard_normal((n_data, 1))
+    Y = signal + noise
+    return X, Y
 
 
-def generate_data(N=80):
-    X = np.random.rand(N)[:, None] * 10 - 5  # Inputs, shape N x 1
-    F = 2.5 * np.sin(6 * X) + np.cos(3 * X)  # Mean function values
-    NoiseVar = 2 * np.exp(-((X - 2) ** 2) / 4) + 0.3  # Noise variances
-    Y = F + np.random.randn(N, 1) * np.sqrt(NoiseVar)  # Noisy data
-    return X, Y, NoiseVar
+X, Y = generate_data()
 
+# %% [markdown]
+# The data alone looks like:
 
-X, Y, NoiseVar = generate_data()
+# %%
+plot_distribution(X, Y)
 
 # %% [markdown]
-# Here's a plot of the data, with error bars representing two standard deviations:
+# ### Try a naive fit
+# If we try to fit a naive GPR to this we get:
 
 # %%
-fig, ax = plt.subplots(1, 1, figsize=(12, 6))
-_ = ax.errorbar(
-    X.squeeze(),
-    Y.squeeze(),
-    yerr=2 * (np.sqrt(NoiseVar)).squeeze(),
-    marker="x",
-    lw=0,
-    elinewidth=1.0,
-    color="C1",
+model = gf.models.GPR(
+    data=(X, Y),
+    kernel=gf.kernels.SquaredExponential(),
 )
-
-# %% [markdown]
-# ### Make a Y matrix that includes the variances
-# We need to tell the GP model what the variance is for each data point. To do this, we'll concatenate the observations with the variances into a single data matrix:
+gf.optimizers.Scipy().minimize(
+    model.training_loss, model.trainable_variables, options=optimizer_config
+)
+print_summary(model)
 
 # %%
-Y_data = np.hstack([Y, NoiseVar])
+mean, var = model.predict_y(X_plot)
+plot_distribution(X, Y, X_plot, mean.numpy(), var.numpy())
 
+# %% [markdown]
+# Notice how this naive model underestimates the noise near the center of the figure, but overestimates at for small and large values of x?
 
 # %% [markdown]
-# ### Make a new likelihood
+# ### Fit a polynomial to the noise scale
 #
-# To cope with this data structure, we'll build a new likelihood. Note how the code extracts the observations `Y` and the variances `NoiseVar` from the data. For more information on creating new likelihoods, see [Likelihood design](../tailor/likelihood_design.ipynb). Here, we're implementing the `log_prob` function (which computes the log-probability of the data given the latent function) and `variational_expectations`, which computes the expected log-probability under a Gaussian distribution on the function, and is needed in the evaluation of the evidence lower bound (ELBO). Check out the docstring for the `Likelihood` object for more information on what these functions do.
+# To fit a model with varying noise we first create a `Likelihood` with a (polynomial) `Function` for the noise,
```

Comment:
```suggestion
# To fit a model with varying noise, instead of using the default single shared noise variance, we can create a Gaussian `Likelihood` with a (polynomial) `Function` for the noise variance,
```